app/controllers/staff/activity_forms_controller.rb

```diff
@@ -31,14 +31,42 @@ def update
     @activity = Activity.find(params[:activity_id])
     authorize @activity
 
-    @activity.assign_attributes(activity_params)
+    # Update activity dates, after checking they are valid via format_date
+    update_activity_dates
+    # Update all attributes except dates
+    @activity.assign_attributes(activity_params.select { |param| param.match(/^((?!date).)*$/) })
+
     update_wizard_status
 
     render_wizard @activity
   end
 
   private
 
+  def update_activity_dates
+    return if activity_params.select { |param| param.match(/(.*)date(.*)/) }.empty?
```

Comment:
Actually, would it be better to drop the regex and guard each update with the presence of the matching param?

Reply:
If I do that it's going to be something like 

```
@activity.update(planned_start_date: format_date(planned_start_date_params)) if [activity_params["planned_start_date(1i)"],activity_params["planned_start_date(2i)"],activity_params["planned_start_date(3i)"]].any?
```
because there's some Rails magic going on with the date params